### ISSUE TITLE
Shipping Method Always Defaulting To Local Pickup

### ISSUE THREAD

**Issue by byronyasgur** [woothemes/woocommerce#7871]
I'm still getting this problem detailed in https://github.com/woothemes/woocommerce/issues/7721 , and I've tested it extensively ... it always defaults to local pickup if free shipping is enabled but does not qualify; and flat rate is not chosen as the default.

Please see this video created with a new WordPress install, no other plugins, default settings except for the shipping setup. As I show at the start of the video I put the system in Shopping Debug Mode and clear sessions, and I'm in chrome incognito mode with no extensions enabled.

https://www.youtube.com/watch?v=S2sLiMC9suc **( skip to 0:38 to see bug demonstration )**

As you can see from the video **it works fine until I enable local pickup, then it defaults to that no matter what is chosen in the default radio.** unless free shipping has qualified.

**Comment by mikejolley** [comment]
You can only set 1 default, so if your default is non-existant, the new default will be cheapest method. i.e. local pickup.

**Comment by byronyasgur** [comment]
By 'non existent' I'm assuming you mean that free shipping doesn't qualify ( as opposed to it not being set as a default).

In that case have you considered that it's impossible then to setup a shop properly if the client requires both free shipping and local pickup methods, because the developer is forced to choose between either  __A) Set 'free shipping' as default__ which will mean it defaults to local pickup if free shipping doesn't qualify  ... or __B) Set 'Flat Rate' as default__ which means Free shipping will not be automatically selected if the cart qualifies.

__The Catch 22__ Customers will not always pay proper attention to the selected method unless they require non standard shipping; but if I go with option A, then they'll be waiting for the product to arrive by post while it waits at the shop for them, and if I go with B then they may end up paying flat rate ( if I set __that__ to default ) and the shop may need to organize a refund ... both customer service problems that many clients will not accept.

__My opinion in case it's of any help__ I think most end users will expect the system set up so that the cheapest option will be chosen and you have the option for that; but in my opinion this should exclude include the Local Pickup method; as it not usually make sense to default to that; and if someone needs it as a default they could set it manually anyway.

**Comment by dirkschrama** [comment]
This is exactly the problem I have. It drives me crazy.
1. People want to pick up locally.
2. People have to pay for shipping small orders.
3. People want and get free shipping with bigger order.

Problem => people are in hurry and choose pick-up locally. Then they don't get their stuff and get angry.

**Comment by bootmintheme** [comment]
This still appears to be an issue. 

In /cart/, I have "Local pickup" enabled and "2 shipping options" from a provider (Aramex). 

When I select a shipping option of the provider which has a monetary value, and realise I've entered the wrong address moments before, then proceed to "Calculate shipping" again and change the address, I click on "update totals", it correctly fetches the totals supplied by the provider (Aramex) in this case. 

![image](https://user-images.githubusercontent.com/5621497/96559315-2a232f80-12bd-11eb-8590-a5f622e710be.png)

and it gives the notification "Shipping costs updated".

![image](https://user-images.githubusercontent.com/5621497/96559375-42934a00-12bd-11eb-8134-81d46027f8ce.png)


BUT: Shipping switches back to "local pickup" and does not keep it on the Aramex shipping option or delivery fee selected previously before I did my new shipping calculation; and if the client does not concentrate, places the order without paying for the shipping, since local pick up was defaulted to again after the .cart_totals.calculated_shipping box updated. See below.

![image](https://user-images.githubusercontent.com/5621497/96559640-9aca4c00-12bd-11eb-928b-c61351666660.png)
